File: src/server/TypescriptTest/TestClass.ts
export default class TestClass {
    public constructor( public counter: number ) { }

    
    public increment(count: number = 1) {
        this.counter += count;
    }


    public isPrime() {
        for (let i = 2; i <= math.ceil(this.counter/2); i++) {
            if (this.counter % i === 0) {
                return false;
            }
        }
        
        return true;
    }
}

File: src/server/LuauTest/TestClass.lua
local TestClass = {}
TestClass.__index = TestClass


function TestClass:increment(count: number)
	count = count or 1
	self.counter += count
end


function TestClass:isPrime()
	for i=2, math.ceil(self.counter / 2), 1 do
		if (self.counter % i == 0)  then
			return false
		end
	end
	
	return true
end


function TestClass.new(counter)
	local self = setmetatable({
		counter = counter;
	}, TestClass)
	
	return self
end


return TestClass